Integrations/JsonWhoIs/JsonWhoIs.py

```diff
@@ -0,0 +1,186 @@
+    :param cur_list: list of dict
+    :param key_needed: key to save
+    :return: modified list by description above
+    """
+    if not cur_list:
+        return None
+    cur_list = [createContext(index, removeNull=True) for index in cur_list]
+
+    def cur_ec(index):
+        return {key.capitalize(): index[key] for key in index if key in key_needed}
+    cur_list = [cur_ec(contact) for contact in cur_list]
+    return cur_list
+
+
+''' COMMANDS + REQUESTS FUNCTIONS '''
+
+
+@logger
+def whois(url: str) -> tuple:
+    """Get Rest API raw from JsonWhoIs service
+    :param url: url to search on
+    :return: dict object of JsonWhoIs service
+    """
+    # Perform request
+    params = {
+        'domain': url
+    }
+    raw = http_request(method='GET',
+                       url_suffix='/api/v1/whois',
+                       params=params)
+
+    # entry context by code convention
+    ec_temp = {
+        'DomainStatus': raw.get('status'),
+        'NameServers': list_by_ec(raw.get('nameservers'), key_needed=['name']),
+        'CreationDate': raw.get('created_on'),
+        'UpdatedDate': raw.get('updated_on'),
+        'ExpirationDate': raw.get('expires_on'),
+        'Registrar': dict_by_ec(raw.get('registrar')),
+        'Registrant': list_by_ec(raw.get('registrant_contacts'), key_needed=['name', 'phone', 'email']),
+        'Admin': list_by_ec(raw.get('admin_contacts'), key_needed=['name', 'phone', 'email'])
+    }
+
+    # Clear unused fields
+    ec_temp = createContext(ec_temp, removeNull=True)
+
+    # Build all ec
+    ec = {
+        'Domain': {
+            'WHOIS': ec_temp
+        }
+    }
+
+    return ec, raw
+
+
+@logger
+def whois_command():
+    """Whois command"""
+    # Get url arg
+    url = demisto.args().get('query')
+    # Get parsed entry context and raw data
+    ec, raw = whois(url)
+
+    # Create human-readable format
+    ec_shortcut = ec['Domain']['WHOIS']
+    human_readable = ''
+    if 'Admin' in ec_shortcut:
+        human_readable += tableToMarkdown(name='Admin account', t=ec_shortcut['Admin'])
+    if 'NameServers' in ec_shortcut:
+        human_readable += tableToMarkdown(name='Name servers', t=ec_shortcut['NameServers'])
+    if 'Registrant' in ec_shortcut:
+        human_readable += tableToMarkdown(name='Registrant', t=ec_shortcut['Registrant'])
+    if 'Registrar' in ec_shortcut:
+        human_readable += tableToMarkdown(name='Registrar', t=ec_shortcut['Registrar'])
+
+    # Others table
+    others_keys = ['DomainStatus', 'CreationDate', 'UpdatedDate', 'ExpirationDate']
+    ec_others = {key: ec_shortcut[key] for key in ec_shortcut if key in others_keys}
+    human_readable += tableToMarkdown(name='Others', t=ec_others)
+
+    return_outputs(readable_output=human_readable,
+                   outputs=ec,
+                   raw_response=raw)
+
+
+@logger
+def test_module():
+    ec, raw = whois('demisto.com')
+    status = ec.get('Domain').get('WHOIS').get('DomainStatus')
+    if not status:
```

Comment:
the test button needs to check that the API works, not that it returns specific values.
if the call to `whois('demisto.com')` worked - that's enough.